hcl/eval/functions.go

```diff
@@ -67,6 +72,52 @@ func tmAbspath(basedir string) function.Function {
 	})
 }
 
+func tmVendor(basedir, vendordir project.Path, stream chan<- event.TmVendorEvent) function.Function {
+	return function.New(&function.Spec{
+		Params: []function.Parameter{
+			{
+				Name: "modsrc",
+				Type: cty.String,
+			},
+		},
+		Type: function.StaticReturnType(cty.String),
+		Impl: func(args []cty.Value, retType cty.Type) (cty.Value, error) {
+			// Param spec already enforce modsrc to be string.
+			source := args[0].AsString()
+			modsrc, err := tf.ParseSource(source)
+			if err != nil {
+				return cty.NilVal, errors.E(err, "tm_vendor: invalid module source")
+			}
+			targetPath := modvendor.TargetDir(vendordir, modsrc)
+			result, err := filepath.Rel(basedir.String(), targetPath.String())
```

Comment:
both paths are UNIX paths then this should not be a failing operation but I didn't find a `path.Rel()`, unfortunately, then panic is our only option =(